native/mod_manager/mod_manager.c

```diff
@@ -1748,23 +1748,29 @@ static char * process_appl_cmd(request_rec *r, char **ptr, int status, int *errt
         return (process_node_cmd(r, status, errtype, node));
     }
 
-    /* Read the ID of the virtual host corresponding to the first Alias */
+    /* Go through the provided Aliases, the first Alias that matches an existing host gets used
+     * otherwise, a new host will be created
+     */
     hostinfo.node = node->mess.id;
+    hostinfo.id = 0;
     if (vhost->host != NULL) {
-        char *s = hostinfo.host;
-        int j = 1;
-        strncpy(hostinfo.host, vhost->host, sizeof(hostinfo.host));
-        hostinfo.host[sizeof(hostinfo.host)-1] = '\0';
-        while (*s != ',' && j<sizeof(hostinfo.host)) {
-           j++;
-           s++;
-        }
-        *s = '\0';
-    } else
+        int start = 0;
+        i = 0;
+        while (host == NULL && i + start < strlen(vhost->host)) {
+            while (vhost->host[start+i] != ',' && vhost->host[start+i] != '\0') {
+                i++;
+            }
+
+            strncpy(hostinfo.host, vhost->host + start, i);
+            hostinfo.host[i] = '\0';
+            host = read_host(hoststatsmem, &hostinfo);
+            start = start + i + 1;
+            i = 0;
+        }
```

Comment:
@jajik Sorry for delay, but I am under suspicion this is still a little wrong / problematic. While the 'matching' fix is correct and should really result in 1 vhost if any of the aliases are overlapping - the result bit should not be a merge, but a replace.

E.g. an existing vhost with aliases `A,B` receives update with `B,C`, the resulting vhost should be `B,C` rather than a merge of `A,B,C`. 

Let me know if I misunderstood.

Reply:
@jfclere What do you think? May that result in some node infighting?

Reply:
@rhusar hm do we want to merge?... because if the back-end send a ENABLE context/hosts, so you are thinking of:
- /test A,B
- /other B, C
that means you have in the back-end
host1 : A, B
host2: B, C
So a request to /test on A should go to test on host1 and a request to /test on A should give a 404 (well look to have context/hosts).

so merging looks wrong that way... B, C also looks wrong no?

Reply:
host1, host2 are jvmroutes and A, B, C aliases in this case?

Reply:
and:
/test A,B (VM1)
/test B,C (VM2)
Should be rejected : we don't know how to route /test B ...

Reply:
> host1, host2 are jvmroutes and A, B, C aliases in this case?

Yes... I have edited the comment to make it understandable.

Reply:
But in case of different JVMRoutes the merge does not happen.

Reply:
@jajik correct.

Reply:
It would create

Node host1
* Contexts
    * /test
* Aliases
    * A, B 

Node host2
 * Contexts
     * /other
* Aliases
    * B, C

Reply:
@rhusar @jajik so code is correct.
VM1 /test A, B
some configuration changes
VM1 /test B, C
We should not route A / test there correct?